File: frontend/lib/types/browser-profile.ts
export interface AnalysisRequest {
  fingerprint_hash: string;
  browser_tier: "modern" | "legacy";
  device_class: "high" | "mid" | "low";
  network_class: "fast" | "slow" | "unknown";
  timezone_region: string;
  language_group: string;
  storage_capabilities: string[];
  uniqueness_score: number;
  behavioral_signals: {
    time_on_page_seconds: number;
    scroll_depth_percent: number;
    interaction_count: number;
    idle_ratio: number;
  };
  raw_data?: Record<string, any>;
}

export interface AIAnalysisResponse {
  ad_profile: {
    inferred_demographics: string;
    income_bracket: "High" | "Medium" | "Low";
    interests: string[];
    purchase_intent_signals: string[];
  };
  recommendation_buckets: Array<{
    platform: string;
    bucket_name: string;
    mechanics: string;
  }>;
  data_valuation: {
    estimated_cpm_usd: number;
    valuation_tier: "Premium" | "Mid-Tier" | "Value";
    factors: string[];
  };
  dark_patterns: Array<{
    name: string;
    description: string;
    severity: "High" | "Medium" | "Low";
  }>;
  protection_advice: Array<{
    action: string;
    impact: string;
    difficulty: "Easy" | "Moderate" | "Hard";
  }>;
  meta: {
    tokens_used: number;
    latency_ms: number;
    scan_id?: string;
  };
}

File: backend/app/schemas/browser_profile.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional

class BehavioralSignals(BaseModel):
    time_on_page_seconds: float = Field(..., description="Seconds spent on the scanner page")
    scroll_depth_percent: float = Field(..., description="Max scroll depth reached by the visitor")
    interaction_count: int = Field(..., description="Count of click, touch, and scroll events")
    idle_ratio: float = Field(..., description="Ratio of idle time to active time")

class AnalysisRequest(BaseModel):
    fingerprint_hash: str = Field(..., description="SHA-256 fingerprint hash")
    browser_tier: str = Field(..., description="modern or legacy")
    device_class: str = Field(..., description="high, mid, or low capability device")
    network_class: str = Field(..., description="fast, slow, or unknown network speed")
    timezone_region: str = Field(..., description=" coarse location/timezone region")
    language_group: str = Field(..., description="Primary language of the visitor")
    storage_capabilities: List[str] = Field(..., description="Detected storage types available")
    uniqueness_score: float = Field(..., description="Uniqueness score from 0 to 100")
    behavioral_signals: BehavioralSignals
    raw_data: Optional[Dict[str, Any]] = Field(default=None, description="All raw collector properties")
